### PR TITLE
Host registration - update 'setup rex & insights'

### PR DESCRIPTION
**Pull request by stejskalleos**
For https://bugzilla.redhat.com/show_bug.cgi?id=1975098

Cherry-pick into:

* [ ] Foreman 2.5 (Satellite 6.10)

### CODE REVIEW

File: guides/doc-Managing_Hosts/topics/proc_registering-a-host.adoc [hunk 1]
```diff
@@ -48,11 +48,21 @@ If an attacker, located in the network between {Project} and a host, fetches the
...
+. From the *Setup REX* list, select whether you want to deploy {Project} SSH keys to hosts or not.
++
+If set to `Yes`, SSH keys will be installed on the registered host. The inherited value is based on the `host_registration_remote_execution` parameter.
+It can be inherited e.g. from host group, operating system, organization. When overridden, the selected value will be stored on host parameter level.
++
 
-ifdef::satellite[]
-. From the *Insights* list, select whether you want to register the hosts to Insights or not.
-endif::[]
+. From the *Setup Insights* list, select whether you want to install `insights-client` & register the hosts to Insights or not.
```
Comment (melcorr) [inline guides/doc-Managing_Hosts/topics/proc_registering-a-host.adoc:57]:
```suggestion
. From the *Setup Insights* list, select whether you want to install `insights-client` and register the hosts to Insights.
```
The "or not" is implied in the "whether" :-)

File: guides/doc-Managing_Hosts/topics/proc_registering-a-host.adoc [hunk 2]
```diff
@@ -48,11 +48,21 @@ If an attacker, located in the network between {Project} and a host, fetches the
...
++
+Insights tool is available for Red Hat family operating systems only, it has no effect on other OS families that do not support it.
+Required enabled repositories on registered machine:
```
Comment (melcorr) [inline guides/doc-Managing_Hosts/topics/proc_registering-a-host.adoc:60]:
```suggestion
You must enable the following repositories on a registered machine:
```

**Review by melcorr — CHANGES_REQUESTED** [review]
Some small nitpicks. 
Do you really think it is worth exposing Insights content upstream? 
I suppose it is no harm with your warning, so I am fine with it.

**Comment by stejskalleos** [comment]
> Do you really think it is worth exposing Insights content upstream?
Yes, the parameter is in foreman-core, so all users can see it. Plus I can imagine there can be cases where users register RHEL machines without `katello` plugin.

**Review by melcorr — APPROVED** [review]
Ok, then. Thank you for making the change!

File: guides/doc-Managing_Hosts/topics/proc_registering-a-host.adoc [hunk 3]
```diff
@@ -48,11 +48,21 @@ If an attacker, located in the network between {Project} and a host, fetches the
...
+If set to `Yes`, SSH keys will be installed on the registered host. The inherited value is based on the `host_registration_remote_execution` parameter.
+It can be inherited e.g. from host group, operating system, organization. When overridden, the selected value will be stored on host parameter level.
++
 
-ifdef::satellite[]
-. From the *Insights* list, select whether you want to register the hosts to Insights or not.
-endif::[]
+. From the *Setup Insights* list, select whether you want to install `insights-client` and register the hosts to Insights.
++
+Insights tool is available for Red Hat family operating systems only, it has no effect on other OS families that do not support it.
```
Comment (ekohl) [inline guides/doc-Managing_Hosts/topics/proc_registering-a-host.adoc:59]: AFAIK it's only available on RHEL, not on the Red Hat family. At least CentOS Stream 8 and CentOS Linux 7 don't have it.
```suggestion
The Insights tool is available for Red Hat Enterprise Linux only. It has no effect on other operating systems.
```

Comment (stejskalleos) [reply guides/doc-Managing_Hosts/topics/proc_registering-a-host.adoc:59]: I shamelessly copy/pasted the text from label help in registration form, guess we should change it there too (and the API too) ...

**Pull request merged**